### PR TITLE
fix(nextjs): remove temporary patch for next eslint rules

### LINKED ISSUE #20777: @nx/eslint: `eslintConfig` config option not working
### Current Behavior

After initialising a new nx nextjs workspace and migrating to eslint flat config i get the following error after setting the `eslintConfig` prop in the lint section of project.json.

> TypeError: Key "rules": Key "@typescript-eslint/no-array-constructor": Could not find plugin "@typescript-eslint".

This only happens when i explicitly link a eslint.config.js file via `eslintConfig` and don't use the root eslint configuration.

### Expected Behavior

Its possible to run the lint task after upgrading to flat config

### GitHub Repo

https://github.com/knoefel/nx-eslint-flat-config

### Steps to Reproduce

1. create new nx workspace
2. run `nx generate @nx/eslint:convert-to-flat-config`
3. update the app config for the lint target by setting a `esLintConfig` property:
```js
    "lint": {
      "executor": "@nx/eslint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "eslintConfig": "apps/lint/eslint.config.js"
      }
    }
```
4. run `nx lint lint`

### Nx Report

```shell
Node   : 20.10.0
   OS     : darwin-arm64
   npm    : 10.2.4
   
   nx (global)        : 17.1.1
   nx                 : 17.2.4
   @nx/js             : 17.2.4
   @nx/jest           : 17.2.4
   @nx/linter         : 17.2.4
   @nx/eslint         : 17.2.4
   @nx/workspace      : 17.2.4
   @nx/devkit         : 17.2.4
   @nx/eslint-plugin  : 17.2.4
   @nx/next           : 17.2.4
   @nx/playwright     : 17.2.4
   @nx/react          : 17.2.4
   @nrwl/tao          : 17.2.4
   @nx/web            : 17.2.4
   typescript         : 5.2.2
```

### Failure Logs

_No response_

### Package Manager Version

_No response_

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by meeroslav**
## Related Issue(s)

Fixes #20777

### CODE REVIEW

File: packages/next/migrations.json [hunk 1]
```diff
@@ -23,6 +23,12 @@
       "version": "16.4.0-beta.3",
       "description": "Update package.json moving @nx/next from dependency to devDependency",
       "implementation": "./src/migrations/update-16-4-0/update-nx-next-dependency"
+    },
+    "update-17-2-7": {
+      "cli": "nx",
+      "version": "17-2-6",
```
Comment (mandarini) [inline packages/next/migrations.json:29]: `17.2.7`?

File: packages/next/src/migrations/update-17-2-7/remove-eslint-rules-patch.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+export default async function update(tree: Tree) {
+  const projects = getProjects(tree);
+  projects.forEach((project) => {
+    updateOverrideInLintConfig(
+      tree,
+      project.root,
+      (o) =>
+        o.rules?.['@next/next/no-html-link-for-pages'] &&
+        o.files.includes('**/*.*'),
```
Comment (mandarini) [inline packages/next/src/migrations/update-17-2-7/remove-eslint-rules-patch.ts:12]: `o.files?.`

**Review by mandarini — APPROVED** [review]
Just a couple of comments!

File: packages/next/src/migrations/update-17-2-7/remove-eslint-rules-patch.ts (hunk 2 above)
Comment (JamesHenry) [reply packages/next/src/migrations/update-17-2-7/remove-eslint-rules-patch.ts:12]: Would be a broken config on the user's side but may as well 🤷

File: packages/eslint/src/generators/utils/eslint-file.ts [hunk 3]
```diff
@@ -207,7 +207,7 @@ function overrideNeedsCompat(
   override: Linter.ConfigOverride<Linter.RulesRecord>
 ) {
   return (
-    !override.env && !override.extends && !override.plugins && !override.parser
+    override.env || override.extends || override.plugins || override.parser
```
Comment (JamesHenry) [inline packages/eslint/src/generators/utils/eslint-file.ts:210]: This seems unrelated?

Comment (meeroslav) [reply packages/eslint/src/generators/utils/eslint-file.ts:210]: As mentioned, removing this temporary patch resurfaced some minor issues in eslint construction which was fixed as a part of this (otherwise the tests would fail).

**Review by JamesHenry — APPROVED** [review]

File: packages/next/migrations.json [hunk 4]
```diff
@@ -23,6 +23,12 @@
       "version": "16.4.0-beta.3",
       "description": "Update package.json moving @nx/next from dependency to devDependency",
       "implementation": "./src/migrations/update-16-4-0/update-nx-next-dependency"
+    },
+    "update-17-2-7": {
+      "cli": "nx",
+      "version": "17-2-7",
```
Comment (mandarini) [inline packages/next/migrations.json:29]: I'm not sure if this syntax (`17-2-7`) has the same effect as the dot syntax (`17.2.7`), but at least for the sake of consistency, should we use `17.2.7` instead?

Comment (meeroslav) [reply packages/next/migrations.json]: Yeah, it doesn't. It was a typo.
